src/flask_smorest/exceptions.py

```diff
@@ -7,30 +7,166 @@ class FlaskSmorestError(Exception):
     """Generic flask-smorest exception"""
 
 
+# Non-API exceptions
 class MissingAPIParameterError(FlaskSmorestError):
     """Missing API parameter"""
 
 
-class NotModified(wexc.HTTPException, FlaskSmorestError):
-    """Resource was not modified (Etag is unchanged)
+class CurrentApiNotAvailableError(FlaskSmorestError):
+    """`current_api` not available"""
 
-    Exception created to compensate for a lack in Werkzeug (and Flask)
-    """
 
-    code = 304
-    description = "Resource not modified since last request."
+# API exceptions
+class ApiException(wexc.HTTPException, FlaskSmorestError):
+    """A generic API error."""
+
+    @classmethod
+    def get_exception_by_code(cls, code):
+        for exception in cls.__subclasses__():
+            if exception.code == code:
+                return exception
+        return InternalServerError
```

Comment:
I think I'd return a `ValueError` here as `code` is wrong (it will result in an internal error anyway).

Reply:
```python
# API exceptions
class ApiException(wexc.HTTPException, FlaskSmorestError):
    """A generic API error."""

    @classmethod
    def get_exception_by_code(cls, code):
        for exception in cls.__subclasses__():
            if exception.code == code:
                return exception
        return ValueError(f"Not a standard HTTP status code: {code}")
```

I think this error message is appropriate, but if you have another one, please let me know.. :)